android/mlc4j/src/main/java/ai/mlc/mlcllm/OpenAIProtocol.kt

```diff
@@ -0,0 +1,111 @@
+package ai.mlc.mlcllm
+
+import kotlinx.serialization.Serializable
+import java.util.*
+
+// Data classes for v1/chat/completions
+// API reference: https://platform.openai.com/docs/api-reference/chat/create
+
+class OpenAIProtocol {
+    @Serializable
+    data class TopLogProbs(
+        val token: String,
+        val logprob: Float,
+        val bytes: List<Int>? = null
+    )
+
+    @Serializable
+    data class LogProbsContent(
+        val token: String,
+        val logprob: Float,
+        var bytes: List<Int>? = null,
+        var top_logprobs: List<TopLogProbs> = listOf()
+    )
+
+    @Serializable
+    data class LogProbs(
+        var content: List<LogProbsContent> = listOf()
+    )
+
+    @Serializable
+    data class ChatFunction(
+        val name: String,
+        var description: String? = null,
+        val parameters: Map<String, String>
+    )
+
+    @Serializable
+    data class ChatTool(
+        val type: String = "function",
+        val function: ChatFunction
+    )
+
+    @Serializable
+    data class ChatFunctionCall(
+        val name: String,
+        // NOTE: arguments should be dict str to any codable
+        // for now only allow string output due to typing issues
+        var arguments: Map<String, String>? = null
+    )
+
+    @Serializable
+    data class ChatToolCall(
+        val id: String = UUID.randomUUID().toString(),
+        val type: String = "function",
+        val function: ChatFunctionCall
+    )
+
+    @Serializable
+    data class ChatCompletionMessage(
+        val role: String,
+        var content: List<Map<String, String>>? = null,
```

Comment:
Introduce a ChatCompletionMessageContent class, with custom serialize method and two constructors that constructs from Text or Parts, reference swift https://github.com/mlc-ai/mlc-llm/blob/main/ios/MLCSwift/Sources/Swift/OpenAIProtocol.swift#L86

Reply:
One way to implement such union class can be

```
data class ChatCompletionMessageContent {
   var kind: enum;
   var data: Any;
};
```

We can use enum to denote the kind, then use helper functions toText, IsText to check the kind